src/Engine/Graphics/DecalBuilder.cpp

```diff
@@ -267,74 +267,71 @@ bool DecalBuilder::ApplyBloodSplatToTerrain(struct Polygon* terrpoly, Vec3f* ter
     unsigned int NumBloodsplats = bloodsplat_container->uNumBloodsplats;
 
     if (NumBloodsplats > 0) {
-        // loop over blood to lay
-        for (uint i = 0; i < NumBloodsplats; ++i) {
-            // get triangle geometery
-            if (uStripType == 4) {
-                WorldXPosR = triverts[0].vWorldPosition.x;  // left
-                WorldXPosL = triverts[3].vWorldPosition.x;  // right
-                WorldYPosU = triverts[1].vWorldPosition.y;  // bott
-                WorldYPosD = triverts[0].vWorldPosition.y;  // top
-            } else if (uStripType == 3) {
-                if (tri_orient) {
-                    WorldXPosR = triverts->vWorldPosition.x;
-                    WorldXPosL = triverts[2].vWorldPosition.x;
-                    WorldYPosU = triverts[1].vWorldPosition.y;
-                    WorldYPosD = triverts[2].vWorldPosition.y;
-                } else {
-                    WorldXPosR = triverts[1].vWorldPosition.x;
-                    WorldXPosL = triverts[2].vWorldPosition.x;
-                    WorldYPosU = triverts[1].vWorldPosition.y;
-                    WorldYPosD = triverts->vWorldPosition.y;
-                }
+        // get triangle geometery
+        if (uStripType == 4) {
+            WorldXPosR = triverts[0].vWorldPosition.x;  // left
+            WorldXPosL = triverts[3].vWorldPosition.x;  // right
+            WorldYPosU = triverts[1].vWorldPosition.y;  // bott
+            WorldYPosD = triverts[0].vWorldPosition.y;  // top
+        } else if (uStripType == 3) {
+            if (tri_orient) {
+                WorldXPosR = triverts->vWorldPosition.x;
+                WorldXPosL = triverts[2].vWorldPosition.x;
+                WorldYPosU = triverts[1].vWorldPosition.y;
+                WorldYPosD = triverts[2].vWorldPosition.y;
             } else {
-                log->Warning("Uknown strip type detected!");
+                WorldXPosR = triverts[1].vWorldPosition.x;
+                WorldXPosL = triverts[2].vWorldPosition.x;
+                WorldYPosU = triverts[1].vWorldPosition.y;
+                WorldYPosD = triverts->vWorldPosition.y;
             }
+        } else {
+            log->Warning("Uknown strip type detected!");
```

Comment:
This should be an assertion?

Reply:
Not a fatal error but should certainly return at that point.

Reply:
I mean, what kind of error this really is? 

If this should never happen, then it should be an Assert / Error. Do we have other strip types? Can they appear here?

Reply:
I think we should differentiate between critical situations which could break gameplay or corrupt saves and something like mild visual glitch. And not asserting left and right on every small thing, warning is enough...

Reply:
can we  settle on debug assert? i appreciate the desire to be 100% error-proof but that should also be balanced by the importance of the logic in question
bloodsplats would be a minor incovenience

Reply:
That's what I'm saying. Nobody reads warnings. Debug assertion at least will be caught and fixed.